src/libcore/ptr.rs

```diff
@@ -512,6 +512,8 @@ impl<T: ?Sized> PartialOrd for *mut T {
 #[unstable(feature = "unique", reason = "needs an RFC to flesh out design")]
 pub struct Unique<T: ?Sized> {
     pointer: NonZero<*const T>,
+    // NOTE: this marker has no consequences for variance, but is necessary
+    // for dropck to understand that we logically contain a `T`.
```

Comment:
Ownership is right; it doesn't matter that Unique doesn't drop its `T` itself -- `dropck` thinks it does.